cernopendata_client/cli.py

```diff
@@ -181,24 +208,29 @@ def download_files(server, recid, doi, title, protocol, expand):
             os.mkdir(path)
         except OSError:
             print("Creation of the directory %s failed" % path)
-    total_files = len(file_locations)
-    for file_location in file_locations:
-        file_name = file_location.split("/")[-1]
-        file_dest = path + "/" + file_name
-        with open(file_dest, "wb") as f:
-            print(
-                "==> Downloading file {} of {}: ./{}/{}".format(
-                    file_locations.index(file_location) + 1,
-                    total_files,
-                    path,
-                    file_name,
-                )
+
+    if name:
+        download_file_locations = get_download_files_by_name(
+            name=name, file_locations=file_locations
+        )
+    elif regexp:
+        download_file_locations = get_download_files_by_regexp(
+            regexp=regexp, file_locations=file_locations
+        )
+    elif range:
```

Comment:
`elif` makes filters mutually exclusive, so one cannot do:

```
$ cernopendata-client download-files --recid 5500 --filter-regexp py --filter-range 1-2
```

This would be useful.

Two solutions:
- document that they are exclusive;
- amend `elif` into a series of filters applied one after another  

In the latter case there is no need to cache anything on the server side, you can simply apply first filter to get the list of files, and another to reduce it further, etc... until there are no more `--filter` arguments.

(Also, beware that somebody might apply the same filter twice or thrice, for example:

```console
$ cernopendata-client download-files --recid 5500 --filter-regexp py --filter-regexp level
```

Reply:
Now, it's working for multiple filters but not for same filter multiple times. I will create a seperate isssue for it.